### PR TITLE
[FLINK-10858] Create Angular Scaffold in Flink Dashboard

### PR DESCRIPTION
**Pull request by vthinkxie**
## What is the purpose of the change

This pull request start to move the Flink Dashboard from angular 1.x to Angular latest version(7.0) 

## Brief change log

Init new angular dashboard which including:

* Angular Scaffold (create new-web-dashboard folder)
* Basic Layout (including sider and empty main content)
* Router module (init empty router)
* Http Interceptor module (handle response error)
* Flink configuration load (load flink config when app initialized)
* npm auto package (add frontend-maven-plugin plugin to build frontend code)

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

You can visit http://localhost:8081/next/index.html after start a standalone cluster or just run `npm run proxy` in `flink-runtime-web/new-web-dashboard`, more detail could be found in `flink-runtime-web/new-web-dashboard/README.md`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by vthinkxie** [comment]
cc @tillrohrmann

**Comment by rmetzger** [comment]
If I just do a `mvn clean package -DskipTest`, can I use this new UI already?
When I request `http://localhost:8081/next/index.html` I get `{"errors":["Unable to load requested file /next/index.html."]}`

**Comment by rmetzger** [comment]
There's probably something wrong in the Flink backend?
```
2019-02-25 19:46:54,478 ERROR org.apache.flink.runtime.rest.handler.legacy.files.StaticFileServerHandler  - error while responding
java.nio.file.FileSystemException: /private/var/folders/js/yfk_y2450q7559kygttykwk00000gn/T/flink-web-7e306a82-200a-401e-a5ed-73d45d83dc30/flink-web-ui/next/index.html: Not a directory
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:91)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
	at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
	at java.nio.file.Files.newOutputStream(Files.java:216)
	at java.nio.file.Files.copy(Files.java:3016)
	at org.apache.flink.runtime.rest.handler.legacy.files.StaticFileServerHandler.respondToRequest(StaticFileServerHandler.java:176)
	at org.apache.flink.runtime.rest.handler.legacy.files.StaticFileServerHandler.respondAsLeader(StaticFileServerHandler.java:143)
	at org.apache.flink.runtime.rest.handler.RedirectHandler.lambda$null$0(RedirectHandler.java:142)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:736)
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:442)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at java.lang.Thread.run(Thread.java:748)
```
In my `/private/var/folders/js/yfk_y2450q7559kygttykwk00000gn/T/flink-web-7e306a82-200a-401e-a5ed-73d45d83dc30/flink-web-ui/` there's only an empty file called `next`.
The `next` folder has been properly created in the flink-dist jar.

File: flink-runtime-web/new-web-dashboard/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+Flink Runtime Web
+=================
+[![Travis branch](https://travis-ci.com/vthinkxie/flink-runtime-web.svg?branch=master)](https://travis-ci.com/vthinkxie/flink-runtime-web)
```
Comment (rmetzger) [inline flink-runtime-web/new-web-dashboard/README.md]: Can you remove this line from the file? once merged to Flink, this build status will be invalid.

File: flink-runtime-web/new-web-dashboard/src/app/share/customize/layout/layout.component.html [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          [nzTitle]="statusService.configuration['flink-revision']"
+          nzPlacement="bottomRight"
+          class="text version">
+          <strong>VERSION:</strong>
+          {{ statusService.configuration['flink-version'] }}
+        </span>
+        <nz-divider nzType="vertical"></nz-divider>
+        <a target="_blank"
+          nz-tooltip
+          nzTitle="Help Document"
```
Comment (rmetzger) [inline flink-runtime-web/new-web-dashboard/src/app/share/customize/layout/layout.component.html]: Maybe rename this to "Apache Flink Documentation"

**Comment by rmetzger** [comment]
Overall, the new UI looks very nice. I'm very excited about this change :) 
image

**Comment by rmetzger** [comment]
@zentol This change adds the following dependencies 
```
@angular-devkit/build-angular 0.10.7 : MIT License
@angular/animations 7.0.4 : MIT License
@angular/cdk 7.3.3 : MIT License
@angular/cli 7.0.7 : MIT License
@angular/common 7.0.4 : MIT License
@angular/compiler 7.0.4 : MIT License
@angular/compiler-cli 7.0.4 : MIT License
@angular/core 7.0.4 : MIT License
@angular/forms 7.0.4 : MIT License
@angular/http 7.0.4 : MIT License
@angular/language-service 7.0.4 : MIT License
@angular/platform-browser 7.0.4 : MIT License
@angular/platform-browser-dynamic 7.0.4 : MIT License
@angular/router 7.0.4 : MIT License
@ant-design/icons-angular 2.0.0-beta.1 : MIT License
@types/jasmine 2.8.16 : MIT License
@types/jasminewd2 2.0.6 : MIT License
@types/lodash 4.14.121 : MIT License
@types/node 8.9.5 : MIT License
ant-design-palettes 1.1.3 : MIT License
classList.js 1.1.20150312 : Public Domain
codelyzer 4.5.0 : MIT License
Commander.js 2.19.0 : MIT License
core-js v2.6.5 : MIT License
d3-array 1.2.4 : BSD 3-clause "New" or "Revised" License
d3-axis 1.0.12 : BSD 3-clause "New" or "Revised" License
d3-brush 1.0.6 : BSD 3-clause "New" or "Revised" License
d3-chord 1.0.6 : BSD 3-clause "New" or "Revised" License
d3-collection 1.0.7 : BSD 3-clause "New" or "Revised" License
d3-color 1.2.3 : BSD 3-clause "New" or "Revised" License
d3-contour 1.3.2 : BSD 3-clause "New" or "Revised" License
d3-dispatch 1.0.5 : BSD 3-clause "New" or "Revised" License
d3-drag 1.2.3 : BSD 3-clause "New" or "Revised" License
d3-dsv 1.1.1 : BSD 3-clause "New" or "Revised" License
d3-ease 1.0.5 : BSD 3-clause "New" or "Revised" License
d3-fetch 1.1.2 : BSD 3-clause "New" or "Revised" License
d3-force 1.2.0 : BSD 3-clause "New" or "Revised" License
d3-format 1.3.2 : BSD 3-clause "New" or "Revised" License
d3-geo 1.11.3 : BSD 3-clause "New" or "Revised" License
d3-hierarchy 1.1.8 : BSD 3-clause "New" or "Revised" License
d3-interpolate v1.3.2 : BSD 3-clause "New" or "Revised" License
d3-path 1.0.7 : BSD 3-clause "New" or "Revised" License
d3-polygon v1.0.5 : BSD 3-clause "New" or "Revised" License
d3-quadtree 1.0.6 : BSD 3-clause "New" or "Revised" License
d3-random 1.1.2 : BSD 3-clause "New" or "Revised" License
d3-scale 2.2.2 : BSD 3-clause "New" or "Revised" License
d3-scale-chromatic 1.3.3 : BSD 3-clause "New" or "Revised" License
d3-selection v1.4.0 : BSD 3-clause "New" or "Revised" License
d3-shape 1.3.4 : BSD 3-clause "New" or "Revised" License
d3-time 1.0.11 : BSD 3-clause "New" or "Revised" License
d3-time-format 2.1.3 : BSD 3-clause "New" or "Revised" License
d3-timer v1.0.9 : BSD 3-clause "New" or "Revised" License
d3-transition v1.2.0 : BSD 3-clause "New" or "Revised" License
d3-voronoi v1.1.4 : (BSD 3-clause "New" or "Revised" License AND MIT License)
d3-zoom 1.7.3 : BSD 3-clause "New" or "Revised" License
D3.js 5.9.1 : BSD 3-clause "New" or "Revised" License
date-fns 1.30.1 : MIT License
iconv-lite v0.4.24 : MIT License
jasmine-spec-reporter 4.2.1 : Apache License 2.0
karma-chrome-launcher 2.2.0 : MIT License
karma-coverage-istanbul-reporter 2.0.5 : MIT License
karma-jasmine v1.1.2 : MIT License
karma-jasmine-html-reporter 0.2.2 : MIT License
karma-runner karma v3.0.0 : MIT License
Lo-Dash 4.17.11 : MIT License
ng-zorro-antd 1.9.0-rc.0 : MIT License
parse5 5.1.0 : MIT License
pivotal's jasmine 2.99.1 : MIT License
Protractor 5.4.2 : MIT License
rw 1.3.3 : BSD 3-clause "New" or "Revised" License
RxJS 6.3.3 : Apache License 2.0
safer-buffer 2.1.2 : MIT License
TinyColor 1.4.1 : MIT License
ts-node 7.0.1 : MIT License
tslib 1.9.3 : Apache License 2.0
tslint 5.11.0 : Apache License 2.0
TypeScript 3.1.6 : Apache License 2.0
Zone.js v0.8.29 : MIT License
```

I don't see anything problematic in there.

**Comment by vthinkxie** [comment]
@rmetzger 
thanks for your review, I will submit another pr including all feature and commits

**Comment by vthinkxie** [comment]
I just submit all commits in #8016 
close this pr now

**Comment by rmetzger** [comment]
Thanks a lot!

**Pull request closed without merging**